libaegisub/common/character_count.cpp

```diff
@@ -65,10 +67,25 @@ size_t count_in_range(Iterator begin, Iterator end, int mask) {
 			UChar32 c;
 			int i = 0;
 			U8_NEXT_UNSAFE(begin + pos, i, c);
-			if ((U_GET_GC_MASK(c) & mask) == 0)
-				++count;
+            if ((U_GET_GC_MASK(c) & mask) == 0) {
+                if (mask & U_GC_Z_MASK) {
+                    UChar32 *result = std::find(std::begin(ass_special_chars), std::end(ass_special_chars), c);
+                    if (result != std::end(ass_special_chars)) {
+                        UChar32 c2;
+                        i = 0;
+                        U8_PREV_UNSAFE(begin + pos, i, c2);
```

Comment:
This is UB if pos is 0, and using U8_PREV is unnecessary. Comparing a trailing byte from a multibyte character to `'\\'` will never produce a false-positive.

Reply:
The problem is that size_t is unsigned and the string could potentially start with \N. I've just put in a check for the position instead to ensure this isn't run on the first character.